spec/reek/examiner_spec.rb

```diff
@@ -108,5 +108,42 @@
         expect(examiner.smells).to eq([])
       end
     end
+
+    context 'with an incomprehensible source that causes Reek to crash' do
+      let(:subject) do
+        examiner = described_class.new 'class C; def does_crash_reek; end; end'
+        allow(examiner).to receive(:smell_repository) do
```

Comment:
>> Can we avoid this by either injecting SmellRepository

I'd prefer that as well but our current Examiner constructor signature is not supporting so we'd have to add this. Which would mean that it would take 4 arguments, which I find quite a lot but still manageable. Are you suggesting to update the Examiner constructor to be like this:

```Ruby

    def initialize(source,
                   filter_by_smells: [],
                   configuration: Configuration::AppConfiguration.default,
                   smell_repository: Smells::SmellRepository.new)
end
```

?

>> or stubbing SmellRepository.new directly?

I can't see the benefit. For me this would be even worse since it is more opaque - you're stubbing something that is not even called directly thus making it harder to build up a mental image. Why do you think this is better?